lib/liquid/variable.rb

```diff
@@ -85,12 +85,25 @@ def render(context)
       end
 
       obj = context.apply_global_filter(obj)
-
       taint_check(context, obj)
-
       obj
     end
 
+    def render_to_output_buffer(context, output)
+      obj = render(context)
+
+      if obj.is_a?(Array)
+        output << obj.join
+      elsif obj.nil?
+      elsif !obj.is_a?(String)
+        output << obj.to_s
+      else
+        output << obj
+      end
```

Comment:
```rb
      elsif !obj.is_a?(String)
        output << obj.to_s
      else
        output << obj
      end
```

could be

```rb
      else
        output << obj.to_s
      end
```

Were you wanting to avoid duplication if we get a string subclass?
https://github.com/ruby/ruby/blob/cf930985da1ae15b66577ec8286e54ad04ccda78/string.c#L5765-L5772